lib/media/presentation_timeline.js

```diff
@@ -219,6 +220,46 @@ shaka.media.PresentationTimeline = class {
     return this.presentationDelay_;
   }
 
+  /**
+   * Gives PresentationTimeline a Stream's timeline so it can size and position
+   * the segment availability window, and account for missing segment
+   * information.
+   *
+   * @param {!Array.<!shaka.media.PresentationTimeRange>} timeline
```

Comment:
You're actually passing an array of `shaka.dash.MpdUtils.TimeRange`, which is a superset of `PresentationTimeRange`.  Please update.